docs/userguides/extensions.md

```diff
@@ -0,0 +1,82 @@
+# Write custom extension scripts using the Code42 CLI and py42
+
+While the Code42 CLI aims to provide an easy way to automate many common Code42 tasks, there will likely be times when you need to script something the CLI doesn't have out-of-the-box.
+
+To accommodate for those scenarios, the Code42 CLI exposes a few helper objects in the `code42cli.extensions` module that make it easy to write custom scripts with `py42` that use features of the CLI (like profiles) to reduce the amount of boilerplate needed to be productive.
+
+## Before you begin
+
+The Code42 CLI is a python application written using the [click framework](https://click.palletsprojects.com/en/7.x/), and the exposed extension objects are custom `click` classes. A basic knowledge of how to define `click` commands, arguments, and options is required.
```

Comment:
I am a fan of line breaks in markdown because of horizontal scrolling in text editors. It seems like most markdown renderers can handle converting the newlines to spaces too. Maybe this is the issue with my IDE -- but I have seen markdown rules around line length too.

Reply:
Huh, for some reason I had it in my mind that the newlines could cause odd linebreaks in the rendered sphinx text, but I tested it out and it doesn't seem to matter. Made it much more readable now!